libparsec/crates/client/tests/unit/certificates_ops/store.rs

```diff
@@ -0,0 +1,437 @@
+    let env = env.customize(|builder| {
+        builder.new_user("bob");
+        builder.update_user_profile("bob", UserProfile::Admin);
+    });
+    let alice = env.local_device("alice@dev1");
+    let store = certificates_store_factory(&env, &alice).await;
+
+    let write = store.for_write().await;
+    let (alice_update_certif, alice_update_signed) = env.get_last_user_update_certificate("bob");
+    let alice_update_certif = AnyArcCertificate::UserUpdate(alice_update_certif);
+
+    // Check that certificate is absent
+    let err = write.get_any_certificate(1).await.unwrap_err();
+    p_assert_matches!(err, GetCertificateError::NonExisting);
+
+    // Add new certificate
+    write
+        .add_next_certificate(1, &alice_update_certif, &alice_update_signed)
+        .await
+        .unwrap();
+
+    // Check that certificate is present
+    let got = write.get_any_certificate(1).await.unwrap();
+
+    p_assert_eq!(got, alice_update_certif);
+
+    store.stop().await;
+
+    // Ensure we don't rely on a cache but on data in persistent database
+    let store = certificates_store_factory(&env, &alice).await;
+    let write = store.for_write().await;
+    let got = write.get_any_certificate(1).await.unwrap();
+
+    p_assert_eq!(got, alice_update_certif);
+}
+
+#[parsec_test(testbed = "empty")]
+async fn add_sequester_authority_certificate(env: &TestbedEnv) {
+    let env = env.customize(|builder| {
+        builder
+            .bootstrap_organization("alice")
+            .and_set_sequestered_organization();
+    });
+    let alice = env.local_device("alice@dev1");
+    let store = certificates_store_factory(&env, &alice).await;
+
+    let write = store.for_write().await;
+
+    let (authority_certif, authority_signed) = env.get_sequester_authority_certificate();
+    let authority_certif = AnyArcCertificate::SequesterAuthority(authority_certif);
+
+    // Check that certificate is absent
+    let err = write.get_any_certificate(1).await.unwrap_err();
+    p_assert_matches!(err, GetCertificateError::NonExisting);
+
+    // Add new certificate
+    write
+        .add_next_certificate(1, &authority_certif, &authority_signed)
+        .await
+        .unwrap();
+
+    // Check that certificate is present
+    let got = write.get_any_certificate(1).await.unwrap();
+
+    p_assert_eq!(got, authority_certif);
+
+    store.stop().await;
+
+    // Ensure we don't rely on a cache but on data in persistent database
+    let store = certificates_store_factory(&env, &alice).await;
+    let write = store.for_write().await;
+    let got = write.get_any_certificate(1).await.unwrap();
+
+    p_assert_eq!(got, authority_certif);
+}
+
+// TODO: CertificatesStore doesn't work with SequesterServiceCertificate for
+// the moment, because it is signed with SequesterSigningKeyDer and it try to
+// unsecure_load with VerifyKey, which is wrong. But SequesterVerifyKey need to
+// know the size of the key, because the signature size depends on.
+#[ignore]
```

Comment:
Replace with a should_panic ?

Reply:
No it shouldn't panic, it should pass but this is a bug

Reply:
:open_mouth: